src/hotspot/share/runtime/lockStack.inline.hpp

```diff
@@ -46,11 +46,11 @@ inline bool LockStack::can_push() const {
   return to_index(_top) < CAPACITY;
 }
 
-inline bool LockStack::is_self() const {
-  Thread* thread = Thread::current();
-  bool is_self = &JavaThread::cast(thread)->lock_stack() == this;
-  assert(is_self == (get_thread() == thread), "is_self sanity");
-  return is_self;
+inline bool LockStack::is_owning_thread() const {
+  JavaThread* thread = JavaThread::current();
+  bool is_owning = &JavaThread::cast(thread)->lock_stack() == this;
+  assert(is_owning == (get_thread() == thread), "is_owning sanity");
+  return is_owning;
```

Comment:
This is going to require a re-test just to make sure that we don't have
a code path into here from the VMThread when it is doing some
JVM/TI stuff (again...).

Reply:
I don't think so. That code did use JavaThread::cast(thread) before which would have fired. But that means I can leave out the JavaThread::cast() now. Let me do that change.

Reply:
Agreed! I read thru the diffs so fast I missed the "JavaThread::cast(thread)" part of this:

> bool is_self = &JavaThread::cast(thread)->lock_stack() == this;

I'll still do a round of testing on v70 just because more runs are better for
shaking out anything that might be racy...